src/app/edit-product/edit-product.component.ts

```diff
@@ -63,20 +67,17 @@ export class EditProductComponent implements OnInit {
   }
 
   editProductDetails(){
-    if(this.selectProductToEdit){
-      if(this.myForm.valid){
-        this.selectProductToEdit = {
-          id: this.selectProductToEdit.id,
-          ...this.myForm.value
-        }
-        if (this.selectProductToEdit) {
-          this.store.dispatch(EditProduct({ id: this.selectProductToEdit.id, product: this.selectProductToEdit }));
-          alert("Product edited successfully!")
-        }
-      } else {
-        alert("Something went wrong while trying to edit product details.")
-      }
+    this.id = this.route.snapshot.paramMap.get("id") ?? "";
+
+    this.product = {
+      id: parseInt(this.id),
```

Comment:
You didn't need to change this, I think that 
`this.selectProductToEdit = {
    id: this.selectProductToEdit.id,
    ...this.myForm.value
  }`

is the nicer way to do it. But it's fine regardless;